File: src/test/scala/org/zezutom/algorithms/scala/sort/test/InversionCounterTest.scala
package org.zezutom.algorithms.scala.sort.test

import org.junit.Assert._
import org.junit._
import org.zezutom.algorithms.scala.sort.InversionCounter
import org.zezutom.util.scala.TestUtil

class InversionCounterTest {

  val counter: InversionCounter = new InversionCounter

  val testUtil: TestUtil = new TestUtil

  @Test def small_scale_test {
    testUtil.assertSort(3, counter.sortAndCount)
  }

  @Test def large_scale_test {
    assertEquals(2407905288L, counter.sortAndCount(testUtil.readFromFile("IntegerArray.txt")))
  }

}

File: src/test/java/org/zezutom/algorithms/java/sort/InversionCounterTest.java
package org.zezutom.algorithms.java.sort;

import org.junit.Test;
import org.zezutom.util.java.TestUtil;

import java.util.Arrays;

import static org.junit.Assert.assertEquals;

public class InversionCounterTest {

    private final InversionCounter counter = new InversionCounter();

    private final TestUtil testUtil = new TestUtil();

    @Test
    public void small_scale_test() {
        int[] nums = new int[] {2, 4, 1, 3, 5};

        // 3 inversions: (2, 1), (4, 1), (4, 3)
        assertEquals(3, counter.sortAndCount(nums));

        // array is sorted
        assertEquals("[1, 2, 3, 4, 5]", Arrays.toString(nums));
    }

    @Test
    public void large_scale_test() {
        assertEquals(2407905288L, counter.sortAndCount(testUtil.readFromFile("IntegerArray.txt")));
    }

}
